### ISSUE TITLE
No response when deployed

### ISSUE THREAD

**Issue by jeongl** [damienpontifex/azure-function-actix-web#1]
Thanks for the example. I've got this deployed to an azure fn, but there seems to be no response from the default handler. Still working for you?

**Comment by damienpontifex** [comment]
Sorry @jeongl I'm not using this actively. It was more of an experimentation for myself, so could very well have aged and is no longer working

**Comment by jeongl** [comment]
Np 👍 I've had the hardest time with rust frameworks being able to match the routes presented from azure. Warp works, but their body parser seems to fail and is hard to debug. These server packages only seen to fail for me when I deploy them.

**Comment by damienpontifex** [comment]
Hey @jeongl I had a play on the weekend on doing this again with Axum. Repo https://github.com/damienpontifex/azure-function-axum

I got it working locally and in azure functions for http request...still validating the timer trigger and others. 
Take a look e.g. the build command I used

https://github.com/damienpontifex/azure-function-axum/blob/b139fcf994a739322ae854d35f280aa7442af27e/Makefile#L1-L5

And minimal funcignore which worked https://github.com/damienpontifex/azure-function-axum/blob/main/.funcignore

**Comment by damienpontifex** [comment]
Oh, yay, also got the timer trigger happening :) 
```
Timer triggered at FunctionHandler { data: Data { timer: Timer { schedule: Schedule { adjust_for_dst: true }, schedule_status: None, is_past_due: false } }, metadata: Metadata { sys: Sys { method_name: "MyTimer", utc_now: 2022-05-29T02:40:20.021763900Z, rand_guid: "215feb6a-0b6f-4464-a69c-ce6789e35dc9" } } }
```

**Comment by jeongl** [comment]
@damienpontifex i'm still having a hard time to get your samples to run - I'll get back to you regarding why I think it's not working for me when deployed.. part of the weirdness is, it only seems to register routes starting with the trigger prefixes, like `/api/HttpTrigger`. I don't understand why your `/` works when deployed.

What did you do to configure to receive a POST webhook to your function from a timer trigger? I'm using a cosmosdb trigger, and my function runs, but I want to receive the metadata which I can't get.

I don't think I'm parsing it incorrectly, it's just not coming in. I expected it to come in from this blurb of documentation - https://docs.microsoft.com/en-us/azure/azure-functions/functions-custom-handlers#request-payload

**Comment by jeongl** [comment]
This is working for me @damienpontifex 🎉  👍  The `/api/HttpTrigger` and `MyTimer` route both work, but not the `/` path.
I haven't been able to build the actix-web package and deploy it, so this is nice to have as an alternative.

Would you possibly know why running the timer trigger runs once? On startup the `MyTimer` route would be hit. I thought it would trigger and send a POST to that route on schedule

**Comment by jeongl** [comment]
actually i think the timer is running on schedule. I put some printLn! statements for the `my_timer` function, but oddly it only runs once. not sure why

**Comment by jeongl** [comment]
sorry for all the comments and post @damienpontifex. I'm still learning rust and azure at the same time. I got the endpoints to run (timer and cosmosdb trigger). It turns out I wasn't properly deserializing the body.

Last thing I need to figure out is why the triggers run only once. It happens on startup, but while it's running I can't make it trigger again. thoughts? Thanks again for the example. this has really helped me.

**Comment by damienpontifex** [comment]
No worries for all the comments 😄, happy to be learning. 

The `/` route was what I saw azure functions pinging on startup...kind of like a health check or something. Not an actual endpoint I was using in my function. And because the `/` request failed, it deemed the function had failed to start and so didn't send the timer request - that was my conclusion.

And yes on serialisation!! some of the bodies are double serialised and all sorts. I used the binding to `serde_json::Value` and logged it out to see what the body was and test the serialisation. Works a treat as it accepts all JSON. 

Not sure about the run once issue...do you also have app insights hooked up? Could you confirm it only registers once in there as well? I noticed `println` and stdout logging wasn't really showing up much for me
